### ISSUE TITLE
M2.3.3 breaks email sending when name contains non-ASCII char

### ISSUE THREAD

**Issue by elvinristi** [magento/magento2#24902]
### Preconditions (*)

1. Magento M2.3.3

### Steps to reproduce (*)
1. Open registration or contact us page
2. Enter correct details and as a name include some non-ASCII char, i.e some `õäöü`

### Expected result (*)

1. After registration user receives email
2. No errors exist in `var/log/exception.log`

### Actual result (*)

1. User doesn't receive email
2. Log includes Exception with message `Invalid header value detected `
```
[2019-10-07 11:33:57] report.CRITICAL: Invalid header value detected {"exception":"[object] (Magento\\Framework\\Exception\\MailException(code: 0): Invalid header value detected at vendor/magento/module-email/Model/Transport.php:104, Zend\\Mail\\Header\\Exception\\RuntimeException(code: 0): Invalid header value detected at vendor/zendframework/zend-mail/src/Header/HeaderValue.php:112)"}
````

Reasons:
- [ZF2 conforms only to RFC 2822](https://github.com/zendframework/zend-mail/blob/master/src/Header/HeaderValue.php#L65)

Potential fix is to add encoding in `vendor/magento/framework/Mail/Template/TransportBuilder.php`

```diff
--- vendor/magento/framework/Mail/Template/TransportBuilder.php.org	2019-10-07 17:16:16.946791204 +0200
+++ vendor/magento/framework/Mail/Template/TransportBuilder.php	2019-10-07 17:16:44.243320666 +0200
@@ -402,6 +402,7 @@
             (string)$template->getSubject(),
             ENT_QUOTES
         );
+        $this->messageData['encoding'] = $mimePart->getCharset();
         $this->message = $this->emailMessageInterfaceFactory->create($this->messageData);

         return $this;
```

**Comment by elvinristi** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @elvinristi. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @elvinristi, here is your [Magento instance](https://i-24902-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24902-2-3-develop.instances.magento-community.engineering/admin](https://i-24902-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by elvinristi** [comment]
**Confirmed following on `https://i-24902-2-3-develop.instances.magento-community.engineering` (can't see `var/log/` files)**

When including non-ASCII char in Sender Name-fields under Store Email Addresses

![Sender Name settings](https://snaps.vaimo.com/elvin.risti/store_email_name_234B7A0A.png)

(n) and trying to subscribe to Newsletter from the footer then following message will be shown to frontend: `Something went wrong with the subscription `.

(y) When removing special chars from "Sender Name" and re-trying newsletter subscribe, then message `Thank you for your subscription` is presented to frontend.

**Comment by elvinristi** [comment]
Some related:

- https://github.com/zendframework/zend-mail/issues/221
- https://github.com/magento/magento2/issues/22778
- https://github.com/magento/magento2/issues/22103

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-21567` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by elvinristi** [comment]
@engcom-Bravo , just make sure you link together `MC-21567` and [ENGCOM-6034](https://jira.corp.magento.com/browse/ENGCOM-6034) (https://github.com/magento/magento2/pull/24906)

**Comment by Karlasa** [comment]
Fix for that in PR #24906

**Comment by engcom-Bravo** [comment]
Hi @elvinristi. Unfortunately I have no access to MC-21567. But I hope Magento links such things together automatically.

**Comment by elvinristi** [comment]
fixed in https://github.com/magento/magento2/pull/24906

**Comment by elvinristi** [comment]
https://github.com/magento/magento2/pull/24906 merged

**Comment by kovinet** [comment]
I am on Magento 2.3.3 and I still have this issue.

**Comment by andrewkett** [comment]
We've worked around this on Magento 2.3.4 with the following composer patch on magento\framework

`diff --git a/Framework/Mail/Address.php b/Framework/Mail/Address.php
index 18e1a8c72f2..7bf55e9fd3f 100644
--- a/Framework/Mail/Address.php
+++ b/Framework/Mail/Address.php
@@ -32,6 +32,11 @@ class Address
         ?string $email,
         ?string $name
     ) {
+        
+        if (is_string($name) && $name != '') {
+            $name = iconv('UTF-8', 'ASCII//TRANSLIT', $name);
+        }
+
         $this->email = $email;
         $this->name = $name;
     }
`

**Comment by utklasad** [comment]
@andrewkett this fixed the issue for us on Magento 2.3.4. The other fixed mentioned here thats merged (#24906) is not enough of a solution for 2.3.4 (worked great in 2.3.3 since we applied it there). The strange part about this is that all our email communication stopped even if the $name only had a-z chars. But as I said the "hack" in /Framework/Mail/Address.php fixed all email communication.

**Comment by prasad1205** [comment]
@andrewkett   I'm using you are solution but in the emails I'm seen if customer using the ä, ü. ö those are coverted into ?," . because of this my email name is not looking good and meshed up.
Please possible provide the solution.
thanks
